### PR TITLE
Fix/cloning-logic

### PR DESCRIPTION
**Pull request by ivanzusko**
This **PR** fixes the cloning logic and reduces a size of the codebase.

### CODE REVIEW

File: src/store/actions/api.actions.js [hunk 1]
```diff
@@ -240,18 +152,42 @@ export const fetchEndpoint = pathname => async dispatch => {
                     return plugin;
             }
         });
+
         const lens = R.lensPath(['plugins']);
-        const preparedApi = R.set(lens, preparedPlugins, response.data);
+        const cloning = obj => flag => R.dissoc('name', obj);
+        const notCloning = obj => flag => flag === false && obj;
+        const adjustIfCloning = obj => {
+            const makeDecision = R.either(
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: WDYT of return first approach:
```js
return R.either(
  notCloning(obj),
  cloning(obj)
)(!!flag)
```

or

```js
const adjustIfCloning = obj =>
  R.either(
    notCloning(obj),
    cloning(obj)
  )(!!flag)
```

File: src/store/actions/api.actions.js [hunk 2]
```diff
@@ -240,18 +152,42 @@ export const fetchEndpoint = pathname => async dispatch => {
...
             dispatch,
             getEndpointSuccess,
-        )(preparedApi, response.data);
+        )(preparedEndpoint, response.data);
     } catch (error) {
         errorHandler(dispatch)(error);
     }
 };
 
+export const willClone = name => dispatch => {
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: maybe:
```js
export const willClone = name => dispatch =>
  dispatch(fetchEndpoint(name, true));
```

Comment (ivanzusko) [reply src/store/actions/api.actions.js]: Caralho 😅

File: src/store/actions/api.actions.js (hunk 1 above)
Comment (ivanzusko) [reply src/store/actions/api.actions.js]: You are the best what happened to me, caraca 🤣  !!! @rafaeljesus

**Comment by rafaeljesus** [comment]
👍

**Pull request merged**